### ISSUE TITLE
Why the attention mask related to the text prompt is causal?

### ISSUE THREAD

**Issue by qingju-flwls** [jasonppy/VoiceCraft#86]
I noticed in the code, 
---> new:   x_attention_mask = torch.triu(torch.ones(x.shape[1], x.shape[1]), diagonal=1).bool().to(x.device)
---> commented out:      x_attention_mask = torch.zeros(x.shape[1], x.shape[1]).bool().to(x.device)
where the original valle x_attention_mask has been changed from full attention to the causal attention, isn't x supposed to attention all x? 
Thanks.

**Comment by jasonppy** [comment]
Thanks,

I tried both in initial experiments, and the two types of masks don't make a difference

**Comment by qingju-flwls** [comment]
Thanks for the explanation.
